hbase-server/src/main/java/org/apache/hadoop/hbase/monitoring/MonitoredRPCHandlerImpl.java

```diff
@@ -53,11 +53,20 @@ public MonitoredRPCHandlerImpl() {
 
   @Override
   public synchronized MonitoredRPCHandlerImpl clone() {
-    return (MonitoredRPCHandlerImpl) super.clone();
+    MonitoredRPCHandlerImpl clone = (MonitoredRPCHandlerImpl) super.clone();
```

Comment:
Do we need to clone and wrap? Could we not make a new constructor on MonitoredRpcHandlerImpl passing original to copy all files but the call info map that wrap handles?

Reply:
Yes, it's a good idea, I'll change it.

Reply:
Hi, @saintstack , in addition to the call info map, I also add a boolean variant to mark if the MonitoredRPCHandlerImpl is a cloned snapshot of the handler level MonitoredRPCHandlerImpl. So when the UI requests to display the call info map after clone the MonitoredRPCHandlerImpl, every request will has its own snapshot of the MonitoredRPCHandlerImpl according to the request time. And the non-snapshot MonitoredRPCHandlerImpl can generate its own call info map which independent with all the snapshots. Thanks.